acme_diags/derivations/acme.py

```diff
@@ -306,13 +306,27 @@ def netflux6(rsds, rsus, rlds, rlus, hfls, hfss):
 
 # FIXME: C901 'cosp_bin_sum' is too complex (22)
 def cosp_bin_sum(cld, prs_low0, prs_high0, tau_low0, tau_high0):  # noqa
-    """sum of cosp bins to calculate cloud fraction in specified cloud top pressure and
-    cloud thickness bins, input variable has dimention (cosp_prs,cosp_tau,lat,lon)"""
+    """sum of cosp bins to calculate cloud fraction in specified cloud top pressure / height and
+    cloud thickness bins, input variable has dimention (cosp_prs,cosp_tau,lat,lon)/(cosp_ht,cosp_tau,lat,lon)"""
     prs = cld.getAxis(0)
     tau = cld.getAxis(1)
 
     prs_low = prs[0]
     prs_high = prs[-1]
```

Comment:
You can adopt some of your logic for lines 314-335 into a separate function to avoid code duplication and increase extensibility/testability. 

Note, I haven't tested the output of refactored code.

Example (with some refactoring):
---
```python
def adjust_prs_val_units(prs, prs_val: int, prs_val0: int) -> int:
    """Adjust the prs_val units based on the prs.id"""
    # COSP v2 cosp_pr in units Pa instead of hPa as in v1
    # COSP v2 cosp_htmisr in units m instead of km as in v1
    adjust_ids = {"cosp_prs": 100, "cosp_htmsir": 1000}

    if prs_val0:
        if prs.id in adjust_ids.keys() and max(prs) > 1000:
            prs_val = prs_val0 * adjust_ids[prs.id]
        else:
            prs_val = prs_val0

    return prs_val
```
Usage:
---
```python
# Before
    prs_low = prs[0]
    prs_high = prs[-1]

    if prs.id == "cosp_prs":
        if max(prs) > 1000:  # COSP v2 cosp_pr in units Pa instead of hPa as in v1
            if prs_low0:
                prs_low0 = prs_low0 * 100
            if prs_high0:
                prs_high0 = prs_high0 * 100
    if prs.id == "cosp_htmisr":
        if max(prs) > 1000:  # COSP v2 cosp_htmisr in units m instead of km as in v1
            if prs_low0:
                prs_low0 = prs_low0 * 1000
            if prs_high0:
                prs_high0 = prs_high0 * 1000

    if prs_low0:
        prs_low = prs_low0
    if prs_high0:
        prs_high = prs_high0
    if prs_low0 is None and prs_high0 is None:
        prs_lim = "total cloud fraction"

# After
    prs_low = prs[0]
    prs_high = prs[-1]
    prs_lim = None

    prs_low = adjust_prs_val_units(prs, prs_low, prs_low0)
    prs_high = adjust_prs_val_units(prs, prs_high, prs_high0)

    if not prs_low0 and not prs_high0:
        prs_lim = "total cloud fraction"


```

Reply:
I used your codes.